ui-admin/src/study/participants/AdHocEmailModal.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from 'react'
+import { StudyEnvContextT } from '../StudyEnvironmentRouter'
+import Modal from 'react-bootstrap/Modal'
+import LoadingSpinner from 'util/LoadingSpinner'
+import Api, { NotificationConfig } from 'api/api'
+import { failureNotification } from 'util/notifications'
+import { Store } from 'react-notifications-component'
+import Select from 'react-select'
+
+/** modal for letting users send custom emails to seleted participants */
+export default function AdHocEmailModal({ enrolleeShortcodes, onDismiss, studyEnvContext }:
+{enrolleeShortcodes: string[], studyEnvContext: StudyEnvContextT, onDismiss: () => void}) {
+  const [isLoading, setIsLoading] = useState(true)
+  const [configs, setConfigs] = useState<NotificationConfig[]>([])
+  const [selectedConfig, setSelectedConfig] = useState<NotificationConfig | null>(null)
+  const [adHocMessage, setAdHocMessage] = useState('')
+  const [adHocSubject, setAdHocSubject] = useState('')
+  useEffect(() => {
+    Api.findNotificationConfigsForStudyEnv(studyEnvContext.portal.shortcode, studyEnvContext.study.shortcode,
+      studyEnvContext.currentEnv.environmentName).then(result => {
+      setConfigs(result)
+      setIsLoading(false)
+    }).catch(() => {
+      Store.addNotification(failureNotification('Could not load notification configs'))
+    })
+  }, [])
+
+  const sendEmail = () => {
+    if (!selectedConfig) {
+      return
+    }
+    Api.sendAdHocNotification({
+      portalShortcode: studyEnvContext.portal.shortcode,
+      studyShortcode: studyEnvContext.study.shortcode,
+      envName: studyEnvContext.currentEnv.environmentName,
+      enrolleeShortcodes,
+      customMessages: { adHocMessage, adHocSubject },
+      notificationConfigId: selectedConfig.id
+    })
```

Comment:
Any reason not to call `onDismiss()` here? When the modal didn't dismiss, I thought maybe something had gone wrong.